### PR TITLE
user routes on API working with db -> Implement user controller and password hash

### PR DESCRIPTION
**Pull request by ruben1**

### CODE REVIEW

File: server/api/user/userController.js [hunk 1]
```diff
@@ -45,14 +39,26 @@ exports.create = function(req, res, next) {
...
 exports.update = function(req, res, next) {
-  var user = req.params.user;
-  var updates = req.body;
+  var username = req.body.username;
+  var newData = req.body;
+  User.update({username: username}, newData, function(err, numberAffected, raw) {
+    if(err) {
+      res.status(404).end();
+    }
+    res.end();
```
Comment (AndrewSouthpaw) [inline server/api/user/userController.js:60]: Is there an appropriate status code to send?

File: server/api/user/userController.js [hunk 2]
```diff
@@ -63,6 +69,17 @@ exports.update = function(req, res, next) {
...
+  var username = req.params.username;
+  User.findOne({username: username}, function(err, user) {
+    if(!user) {
+      res.status(404).end();
+    } else {
+      User.remove({username: username}, function(err) {
+        if(err) {
+          res.status(500).end();
+        }
+        res.status(200).end();
```
Comment (AndrewSouthpaw) [inline server/api/user/userController.js:81]: Is 200 definitely the right code? Not sure, but maybe there's something more specialized?

**Comment by AndrewSouthpaw** [comment]
Comments added.

**Comment by ruben1** [comment]
Hey, look like there are a lot of status codes depending on the situation. 
I'll leave here a link and I'll check every status code when I finish the screenshot part
http://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete

**Comment by AndrewSouthpaw** [comment]
@ruben1 : added a new issue referencing your TODO. merging this PR, looks good.

**Pull request merged**